packages/osc-ui/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,118 @@
+import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from '@radix-ui/react-icons';
+import * as SelectPrimitive from '@radix-ui/react-select';
+import type { ComponentPropsWithRef, ElementRef, FC } from 'react';
+import React, { useState } from 'react';
+import { useModifier } from '../../hooks/useModifier';
+import { classNames } from '../../utils/classNames';
+import { getFieldError } from '../../utils/getFieldError';
+import { Icon } from '../Icon/Icon';
+import { Label } from '../Label/Label';
+import './select.scss';
+
+type Description = {
+    label?: string;
+    icon?: React.ReactNode;
+};
+export interface Props extends ComponentPropsWithRef<typeof SelectPrimitive.Root> {
+    /**
+     * Description for the Select, can be a Label or an Icon
+     */
+    description?: Description;
+    /**
+     * Sets the custom styles, e.g. "Secondary", "Tertiary"
+     */
+    groupVariants?: string[];
+    /**
+     * A placeholder value for the Select, e.g. "Please Select"
+     */
+    placeholder?: string;
+    /**
+     * A boolean that alerts when form is submitted for error handling
+     * @default false
+     */
+    wasSubmitted?: boolean;
+}
+
+export const Select = React.forwardRef<ElementRef<typeof SelectPrimitive.Trigger>, Props>(
+    (props: Props, forwardedRef) => {
+        const {
+            children,
+            description,
+            disabled,
+            groupVariants,
+            placeholder,
+            required,
+            name,
+            wasSubmitted = false,
+        } = props;
+        const [value, setValue] = useState('');
+
+        const errorMessage = getFieldError(value, required);
+        const displayError = wasSubmitted && errorMessage;
+
+        const modifiers = useModifier('c-select', groupVariants);
+        const selectClasses = classNames('c-select', modifiers);
+
+        const setDescription = (desc: Props['description']) => {
```

Comment:
Could this just be?
```suggestion
        const setDescription = (desc: Description) => {
```
As you've got a Description type set above

Reply:
Yeah good shout. Updated.